File: source/XYO/QuantumScript/LibStdFunction.js
// Quantum Script
// Copyright (c) 2016-2024 Grigore Stefan <g_stefan@yahoo.com>
// MIT License (MIT) <http://opensource.org/licenses/MIT>
// SPDX-FileCopyrightText: 2016-2024 Grigore Stefan <g_stefan@yahoo.com>
// SPDX-License-Identifier: MIT

Function.prototype.bind = function() {
	if(this instanceof Function) {
		var fn_ = this;
		var this_ = arguments[0];
		var args_ = [];
		for(var k = 1; k < arguments.length; ++k) {
			args_[k - 1] = arguments[k];
		};
		return function() {
			return fn_.apply(this_, args_.concat(arguments));
		};
	};
	return undefined;
};

File: source/XYO/QuantumScript/LibStdFunction.cpp
// Quantum Script
// Copyright (c) 2016-2025 Grigore Stefan <g_stefan@yahoo.com>
// MIT License (MIT) <http://opensource.org/licenses/MIT>
// SPDX-FileCopyrightText: 2016-2025 Grigore Stefan <g_stefan@yahoo.com>
// SPDX-License-Identifier: MIT

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include <XYO/QuantumScript/LibStdFunction.hpp>

#include <XYO/QuantumScript/LibStdFunction.Source.cpp>

// #define XYO_QUANTUMSCRIPT_DEBUG_RUNTIME

namespace XYO::QuantumScript {

	namespace LibStdFunction {

		void initExecutive(Executive *executive, void *extensionId) {
			executive->compileStringX(libStdFunctionSource);
		};

	};
};
